runtime/src/proofs.rs

```diff
@@ -1,11 +1,22 @@
+use codec::{Decode, Encode};
 use rstd::vec::Vec;
 
-#[derive(Debug)]
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
 pub struct Proof {
     hash: [u8; 32],
     sorted_hashes: Vec<[u8; 32]>,
 }
 
+impl Proof {
+    pub fn new(hash: [u8; 32], sorted_hashes: Vec<[u8; 32]>) -> Self {
```

Comment:
I think this constructor is not necessary

Reply:
I would like to keep it for easier inits for proofs.